yaml_test.go

```diff
@@ -1054,5 +1072,30 @@ func TestYAMLEnvInterpolationValueConversion(t *testing.T) {
 	v, ok := p.Get("number").TryAsInt()
 	require.True(t, ok)
 	assert.Equal(t, 3, v)
+}
+
+type cartoon struct {
```

Comment:
I need this struct to implement the `encoding.TextUnmarshaller` interface. I don't know how can we define a method inside a function..